contracts/src/v0.8/keystone/ocr/OCR2Base.sol

```diff
@@ -0,0 +1,352 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ConfirmedOwner} from "../../shared/access/ConfirmedOwner.sol";
+import {OCR2Abstract} from "./OCR2Abstract.sol";
+
+/**
+ * @notice Onchain verification of reports from the offchain reporting protocol
+ * @dev For details on its operation, see the offchain reporting protocol design
+ * doc, which refers to this contract as simply the "contract".
+ */
+abstract contract OCR2Base is ConfirmedOwner, OCR2Abstract {
```

Comment:
Is there a difference between this and the one in the shared folder?

Reply:
There are some minor differences, mostly optimizations that were done by the Functions team.